File: HackerRank 30 Days/day10/main.js
function main() {
    const fs = require('fs');
    const n = parseInt(fs.readFileSync(0, 'utf8').trim(), 10);
    const binary = n.toString(2);

    let count = 0;
    let maxCount = 0;

    for (const bit of binary) {
        if (bit === '1') {
            count += 1;
            maxCount = Math.max(maxCount, count);
        } else {
            count = 0;
        }
    }

    console.log(maxCount);
}

main();

File: HackerRank 30 Days/day10/main.cs
using System;

class Solution
{
    static void Main()
    {
        int n = int.Parse(Console.ReadLine() ?? "0");
        string binary = Convert.ToString(n, 2);

        int count = 0;
        int maxCount = 0;

        foreach (char bit in binary)
        {
            if (bit == '1')
            {
                count++;
                maxCount = Math.Max(maxCount, count);
            }
            else
            {
                count = 0;
            }
        }

        Console.WriteLine(maxCount);
    }
}
